src/lib.rs

```diff
@@ -136,6 +140,41 @@ extern "C" fn handle() {
             tamagotchi.sleep();
             msg::reply(TmgEvent::Slept, 0).expect("Error in reply `TmgEvent::Slept`");
         }
+        TmgAction::Transfer(new_owner) => {
+            assert_eq!(
+                msg::source(),
+                tamagotchi.owner,
+                "Only owner can set new_owner"
+            );
+            tamagotchi.owner = new_owner;
+            msg::reply(TmgEvent::Transfer(new_owner), 0)
+                .expect("Error in reply `TmgEvent::Transfer`");
+        }
+        TmgAction::Approve(allowed_account) => {
+            assert_eq!(
+                msg::source(),
+                tamagotchi.owner,
+                "Only owner can set allowed_account"
+            );
+            assert_eq!(
+                allowed_account,
+                tamagotchi.owner,
+                "Please set a new allowed_account"
+            );
+            tamagotchi.allowed_account = Some(allowed_account);
+            msg::reply(TmgEvent::Approve(allowed_account), 0)
+                .expect("Error in reply `TmgEvent::Approve`");
+        }
+        TmgAction::RevokeApproval => {
+            assert_eq!(
+                msg::source(),
+                tamagotchi.owner,
+                "Only owner can revoke approval"
+            );
+            tamagotchi.allowed_account = Some(tamagotchi.owner);
```

Comment:
No need to keep `allowed_account` at all.
```suggestion
            tamagotchi.allowed_account = None;
```